src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -1061,16 +1105,16 @@ void InstanceKlass::initialize_impl(TRAPS) {
     if (is_in_error_state()) {
       DTRACE_CLASSINIT_PROBE_WAIT(erroneous, -1, wait);
       ResourceMark rm(THREAD);
-      const char* desc = "Could not initialize class ";
-      const char* className = external_name();
-      size_t msglen = strlen(desc) + strlen(className) + 1;
-      char* message = NEW_RESOURCE_ARRAY(char, msglen);
-      if (NULL == message) {
-        // Out of memory: can't create detailed error message
-          THROW_MSG(vmSymbols::java_lang_NoClassDefFoundError(), className);
+      Handle cause(THREAD, get_initialization_error(THREAD));
+      CLEAR_PENDING_EXCEPTION; // ignore any OOM here.
```

Comment:
`get_initialization_error()` doesn't throw any error. Is `CLEAR_PENDING_EXCEPTION` really needed here?

Reply:
It does not throw any error anymore.  Removed that line.